master/pkg/ssh/ssh_test.go

```diff
@@ -0,0 +1,33 @@
+package ssh
+
+import (
+	"testing"
+
+	"golang.org/x/crypto/ssh"
+	"gotest.tools/assert"
+
+	"github.com/determined-ai/determined/master/internal/config"
+)
+
+func verifyKeys(t *testing.T, keys PrivateAndPublicKeys) {
+	privateKey, err := ssh.ParsePrivateKey(keys.PrivateKey)
+	assert.NilError(t, err)
+
+	publickKey, _, _, _, err := ssh.ParseAuthorizedKey(keys.PublicKey) //nolint:dogsled
+	assert.NilError(t, err)
+	assert.Equal(t, string(publickKey.Marshal()), string(privateKey.PublicKey().Marshal()))
+}
+
+func TestSSHKeyGenerate(t *testing.T) {
+	keys, err := GenerateKey(config.SSHConfig{KeyType: config.RSAKeyType, RsaKeySize: 512})
```

Comment:
this test is so short it doesn't matter, but usually i'd recommend splitting a test like/using subtests with `t.Run()`/ using table driven tests this up since it really is three separate tests. it's easier for whoever has to fix one of them if it fails to read just the one test that failed... but i also doubt this test will ever fail. this comment is just FYI.

Reply:
Big +1 to table-driven tests and subtests.